soc/arm64/intel_socfpga/agilex/soc.h

```diff
@@ -0,0 +1,41 @@
+/*
+ * Copyright (c) 2021, Intel Corporation. All rights reserved.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#ifndef _SOC_H_
+#define _SOC_H_
+
+#include <arch/cpu.h>
+
+/* FPGA config helpers */
+#define INTEL_SIP_SMC_FPGA_CONFIG_ADDR		0x400000
+#define INTEL_SIP_SMC_FPGA_CONFIG_SIZE		0x2000000
+
+/* Register Mapping */
+#define SOCFPGA_CCU_NOC_REG_BASE		0xf7000000
+
+#define SOCFPGA_MMC_REG_BASE			0xff808000
+
+#define SOCFPGA_RSTMGR_REG_BASE			0xffd11000
+#define SOCFPGA_SYSMGR_REG_BASE			0xffd12000
+
+#define SOCFPGA_L4_PER_SCR_REG_BASE             0xffd21000
+#define SOCFPGA_L4_SYS_SCR_REG_BASE             0xffd21100
+#define SOCFPGA_SOC2FPGA_SCR_REG_BASE           0xffd21200
+#define SOCFPGA_LWSOC2FPGA_SCR_REG_BASE         0xffd21300
+
+#define mmio_write_32(addr, data) \
+	sys_write32((data), (addr))
+
+#define mmio_read_32(addr) \
+	sys_read32((addr))
```

Comment:
Why do you want to define new macros?

Reply:
Hi Carlo,

We will re-use our ATF driver code for Zephyr.  The ATF driver codes are using mmio_X function to write/read. 
We add these macros to map mmio_X to Zephyr compatible function, so that we can re-use most of the code from ATF.

Reply:
That's not actually correct. You are defining these macros in a soc-specific header file, but you are using the macros in a general driver. Now, let's say (I know this is really unlikely) that someone wants to use your clock agilex driver, in that case they cannot do that (or at least the driver fails to compile) because they have no access to the macros because these are defined in a SoC header file.

So, you have two ways of fixing this:
1. Move the macros out of this header file in a more accessible place so that it can be used together with the drivers using the macros
2. Remove the macros entirely

Reply:
Is it acceptable that I move the mmio macro to the beginning of 'clock_agilex_ll.c' itself? 
This allow us to re-use the existing code.

Reply:
Yes, but please add a comment explaining why, otherwise it looks odd :)

Reply:
Hi Carlo,

Noted and thanks.

A side question, I am new to open source upstream process, is that I should close this pull-request now and resubmit a new pull-request after I clean up all codes?

Reply:
Nope, you have to force push your branch with the changes using git `git push --force ....`, that will retrigger the process.

Please also note that besides addressing my comments you have two other issues to address: the compliance and the scancode checks are failing (see the red crosses at the bottom of this PR).

Reply:
Noted and thanks for reminder for the compliance and scancode error. I will check that.